application/src/main/java/org/togetherjava/tjbot/commands/system/CommandProvider.java

```diff
@@ -0,0 +1,61 @@
+package org.togetherjava.tjbot.commands.system;
+
+import org.togetherjava.tjbot.commands.BotCommand;
+import org.togetherjava.tjbot.commands.UserInteractor;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Provides all registered commands.
+ */
+public interface CommandProvider {
+    /**
+     * Gets a list of all currently available and registered interactors.
+     *
+     * @return all interactors
+     */
+    Collection<UserInteractor> getInteractors();
+
+    /**
+     * Gets a list of all currently available and registered interactors.
+     *
+     * @return all interactors
+     */
+    default Collection<BotCommand> getCommands() {
+        return getInteractors().stream()
+            .filter(BotCommand.class::isInstance)
+            .map(BotCommand.class::cast)
+            .toList();
+    }
+
+    /**
+     * Gets the slash command registered under the given name, if any.
+     *
+     * @param name the name of the command (including its prefix)
```

Comment:
an example would help to understand what "prefix" means

"the full name of the command (including its prefix), e.g. `s-ping`"